### PR TITLE
Added tests for text entry on mobile using an IME simulator in flutter_test_robots (Resolves #620)

### LINKED ISSUE #620: Add SuperTextField mobile text entry tests
`SuperTextField` has a number of desktop text input tests, but doesn't have any mobile text entry tests. Add tests for mobile text entry.

Testing `SuperTextField` mobile behavior requires simulated IME text input.

Create an IME simulator and prove it's functionality by writing a handful of Android and iOS `SuperTextField` user interaction tests.

Use IME simulator to write smoke tests for Android and iOS text entry (also expand smoke tests to include things like deleting content back to empty, selecting and deleting a word, a paragraph, etc.)

### PR DESCRIPTION
**Pull request by matthew-carroll**
Added tests for text entry on mobile using an IME simulator in flutter_test_robots (Resolves #620)

### CODE REVIEW

File: super_editor/test/super_textfield/super_textfield_mobile_keyboard_test.dart [hunk 1]
```diff
@@ -1,18 +1,188 @@
...
+          controller.selection = const TextSelection(baseOffset: 3, extentOffset: 10);
+
+          await tester.ime.typeText("f", getter: imeClientGetter);
+
+          expect(SuperTextFieldInspector.findText().text, "-->f<--");
+          expect(SuperTextFieldInspector.findSelection(), const TextSelection.collapsed(offset: 4));
+        });
+      });
+
+      // TODO: implement newline tests when SuperTextField supports configuration of the action button
```
Comment (angelosilvestre) [inline super_editor/test/super_textfield/super_textfield_mobile_keyboard_test.dart:77]: Just one note: when the text field is multiline it already uses `TextInputAction.newline` as its input action.

**Review by angelosilvestre — COMMENTED** [review]
LGTM
